### PR TITLE
removed unneeded slice() calls from frontend components in subdirectories

### PR DESCRIPTION
**Pull request by michaelchadwick**
refs ilios/ilios#5660

Except for a few instances where the data was going to be `sort()`-ed or `reverse()`-ed or similar, and we now know that does not allow for modification yet, I removed all the `slice()` calls I could find.

### CODE REVIEW

File: packages/frontend/app/components/learner-group/calendar.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ export default class LearnerGroupCalendarComponent extends Component {
     }
     const offerings = await all(mapBy(learnerGroups, 'offerings'));
     const flat = offerings.reduce((flattened, obj) => {
-      return [...flattened, ...obj.slice()];
```
Comment (jrjohnson) [inline packages/frontend/app/components/learner-group/calendar.js:50]: Well this is a bad variable name!

File: packages/frontend/tests/integration/components/program-year/list-test.js [hunk 2]
```diff
@@ -96,29 +96,29 @@ module('Integration | Component | program-year/list', function (hooks) {
     await component.newProgramYear.years.select(thisYear);
     await component.newProgramYear.done.click();
     const programYears = await this.owner.lookup('service:store').findAll('program-year');
-    const sortedProgramYears = sortBy(programYears.slice(), 'id');
+    const sortedProgramYears = sortBy(programYears, 'id');
```
Comment (jrjohnson) [inline packages/frontend/tests/integration/components/program-year/list-test.js:99]: I'm a little bit worried about updating the tests and the application code in the same PR here. Are the changes linked? Or could the tests updates be in a second PR?

Comment (michaelchadwick) [reply packages/frontend/tests/integration/components/program-year/list-test.js:99]: Should be linked to the application code here: https://github.com/ilios/frontend/pull/8095/files#diff-12e203d3f8d49a7214eafaf3aa33d10d74239acd9d6ac6acc1bae17df2fc5256

Also, I changed tests alongside app code with my last de-slice()-ing, so... :D

Comment (jrjohnson) [reply packages/frontend/tests/integration/components/program-year/list-test.js:99]: I accept this 🤠

**Review by jrjohnson — APPROVED** [review]

**Pull request merged**